src/main/kotlin/com/hypto/iam/server/service/UsersService.kt

```diff
@@ -89,6 +89,20 @@ class UsersServiceImpl : KoinComponent, UsersService {
         return getUser(ResourceHrn(orgId, "", IamResources.USER, user.username), user)
     }
 
+    override suspend fun changeUserPassword(
+        organizationId: String,
+        userName: String,
+        oldPassword: String,
+        newPassword: String
+    ): BaseSuccessResponse {
+        val org = organizationRepo.findById(organizationId)
+            ?: throw EntityNotFoundException("Invalid organization id name. Unable to authenticate user")
+        val identityGroup = gson.fromJson(org.metadata.data(), IdentityGroup::class.java)
+        identityProvider.authenticate(identityGroup, userName, oldPassword)
+        identityProvider.setUserPassword(identityGroup, userName, newPassword)
```

Comment:
In case `identityProvider.authenticate` call fails because of wrong oldPassword, what exception is thrown? Also, what will the status code of the HTTP response be?

Reply:
It will return `401`